plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -1865,6 +1869,10 @@ def exec_module(self, **kwargs):
                             else:
                                 data_disk_vhd = None
                                 data_disk_managed_disk = self.compute_models.ManagedDiskParameters(storage_account_type=data_disk['managed_disk_type'])
+                                if data_disk.get('disk_encryption_set'):
```

Comment:
Does it mean that 'disk_encryption_set' can only be specified if 'managed_disk_type' is not configured for data_disks?

Reply:
I think so. I'm not an expert at Azure but from what I can see, it is only the class `ManagedDiskParameters` which has the parameter `disk_encryption_set` as per the Azure Python SDK: https://learn.microsoft.com/en-us/python/api/azure-mgmt-compute/azure.mgmt.compute.v2023_07_01.models.manageddiskparameters?view=azure-python

I'm not sure if other types of disk can be encrypted using a DES. That might be the case. But for example my current customer has a use case where they use managed disk and need encryption with CMK. This would solve that use case at least.